jetty-http-tools/src/main/java/module-info.java

```diff
@@ -0,0 +1,28 @@
+//
+// ========================================================================
+// Copyright (c) 1995-2020 Mort Bay Consulting Pty Ltd and others.
+//
+// This program and the accompanying materials are made available under
+// the terms of the Eclipse Public License 2.0 which is available at
+// https://www.eclipse.org/legal/epl-2.0
+//
+// This Source Code may also be made available under the following
+// Secondary Licenses when the conditions for such availability set
+// forth in the Eclipse Public License, v. 2.0 are satisfied:
+// the Apache License v2.0 which is available at
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+// ========================================================================
+//
+
+module org.eclipse.jetty.http.tools
+{
+    exports org.eclipse.jetty.http.tools;
+    exports org.eclipse.jetty.http.tools.matchers;
+
+    requires transitive org.eclipse.jetty.http;
+
+    // Optional - only required if you use hamcrest too.
+    requires static org.hamcrest;
```

Comment:
OMG. Just to support `assertThat()` now we need to depend on an automatic module.
Can we just get rid of the `hamcrest` dependency? It adds _nothing_ and just complicates things -- everything is already available in the `HttpFields` API.

Reply:
I looked into that (getting rid of hamcrest in jetty-http-tools), but it made for a MUCH larger PR to change the existing code (in the rest of jetty) that uses these hamcrest features exposed in jetty-http-tools.
If you are gung-ho on this, I'll start work on removing it.

Reply:
I like the hamcrest style!

Reply:
Alright, will do.
I'll rebuild this PR from scratch with the jetty-http-tools split between top level and matchers still in /tests/